### DISCUSSION TITLE
Heatmap of user-defined peaks and export Gscore Matrix with Magic correction

### DISCUSSION THREAD

**Discussion by wilfridricher** [GreenleafLab/ArchR#860] [Questions / Documentation] [answered]
Dear Ryan and Jeffrey,

I have 2 different questions:
 
1. I would like to create a heatmap that displays user-defined peaks (especially from public peak signatures).
I have processed the custom enrichment (12.4) and custom deviations (13.2) pipelines and sucessed to process the enrichment heatmap and the embedding visualization (and a heatmap of these signatures).
However, in these processes, we are limited to the signatures and I am really interested on visualizing the peaks individually.

2. In addition, I successed to extract GeneScore matrix using getMatrixFromProject() function, and I would like to know if it is possible to extract the same GeneScore matrix with the Impute Weights data (of Magic) or if I need to use Magic package outside ArchR to get them.
 
Thank a lot !
Wilfrid

**Comment by rcorces** [comment, marked as answer]
1. We do not provide code to do this. You would need to figure this out yourself
2. You can apply the impute weights to the gene score matrix or you can run Magic as you mentioned.

**Reply by wilfridricher** [reply]
Ok, thank you for these answers.

**Reply by Zepeng-Mu** [reply]
I would also like to have the feature that plot Z score of peaks or gene scores in a heatmap without the need to use an SE output. Although doing that using custom code is not very hard.
